include/library.h

```diff
@@ -146,6 +147,7 @@ class Library
 {
   std::map<std::string, kiwix::Book> m_books;
   std::map<std::string, std::shared_ptr<Reader>> m_readers;
+  std::map<std::string, std::shared_ptr<zim::Archive>> m_archives;
```

Comment:
@mgautierfr Currently the archive is held as a unique pointer inside the reader, do you think for the time being making it a shared pointer and copy into the above m_archive map when creating instead of directly making it from the path is a good idea?

Reply:
Yes, we should move to a shared_ptr.
I think the "correct" API would be to always creating a reader from a existing archive.
For compatibility issue, we should have a constructor taking a path an create the archive internally.
On the library side, we should also first create the archive and create the Reader using it.